src/c++/perf_analyzer/genai-perf/genai_perf/logging.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+import logging.config
+from pathlib import Path
+
+from genai_perf.constants import DEFAULT_DATE_FORMAT, DEFAULT_LOG_FORMAT
+
+
+def init_logging(log_file: Path) -> None:
```

Comment:
Moving to thread so it can be collapsed when resolved
---
Elias:

> Error output is in the example.
> As for file, stdout, both. The log config controls this.
> Logging is controlled by the logger & handler using the following steps:
> 
> If both the handlers and loggers sections specify a level, the process works as follows:
> 
> The logger checks if the message level is at or above its own level.
> If it is, the message is passed to the handler.
> The handler then checks if the message level is at or above its own level before it decides to output the message.
> There is a logger set for each file that does logging, which is the recommended best practice. Each is set to the console and a file.
> The console is set to process any logging statement info and up.
> The file handler is set to process any logging statement warning and up.
> 
> two examples:
> 
> log.error(Something)
> check logger -> set to debug which is lower than error so handle
> check console -> set to info which is lower than error so handle
> check file -> set to warning which is lower than error so handle
> 
> log.info(Something else)
> check logger -> set to debug which is lower than info so handle
> check console -> set to info which is equal to info so handle
> check file -> set to warning which is higher than info so ignore
> 
> So depending on the level we set for each logger in the config, we can control where a message is output with fine grain control.

Reply:
that is correct
i am happy to adjust it but figured it was a sane starting position to iterate on.

Reply:
My logic being 
debug: we are actively developing, lets show that in the console for rapid dev work
info: these are things we want to raise to a user, lets not make them dive into a file
warn+: ruh roh. something is big time wrong. Lets log the details in a file for debugging. This can also be passed to us from a user if something goes bad out in the wild.

Reply:
You raise a good point. 
I am okay with only using console output for now.
This isnt set in stone.
if we get an ask for a log file option later, we can revisit and extend this feature.
I want @dyastremsky to chime in but i am okay removing file support.

Reply:
I agree. The logic here makes sense for removing it.

Reply:
Removed